packages/vis/src/composables/useTimeline.ts

```diff
@@ -0,0 +1,348 @@
+    if (!timeline.value) return
+    return timeline.value.getEventProperties(event)
+  }
+
+  const getItemRange = () => {
+    if (!timeline.value) return
+    return timeline.value.getItemRange()
+  }
+
+  const getSelection = () => {
+    if (!timeline.value) return
+    return timeline.value.getSelection()
+  }
+
+  const getVisibleItems = () => {
+    if (!timeline.value) return
+    return timeline.value.getVisibleItems()
+  }
+
+  /**
+   * 
+   * Get the current visible window.
+   *
+   */
+  const getWindow = () => {
+    if (!timeline.value) return
+    return timeline.value.getWindow()
+  }
+
+  /**
+   * 
+   * Move the window such that given time is centered on screen.
+   * 
+   * @param time 
+   * @param options
+   * 
+   */
+  const moveTo = (time: Date, options: TimelineAnimationOptions) => {
+    if (!timeline.value) return
+    timeline.value.moveTo(time, options);
+  }
+
+  /**
+   * 
+   * Force a redraw of the Timeline. The size of all items will be 
+   * recalculated. Can be useful to manually redraw when option 
+   * autoResize=false and the window has been resized, or when 
+   * the items CSS has been changed.
+   * 
+   */
+  const redraw = () => {
+    if (!timeline.value) return
+    timeline.value.redraw()
+  }
+
+  /**
+   * 
+   * Remove vertical bars previously added to the timeline via addCustomTime method.
+   * 
+   * @param id — ID of the custom vertical bar returned by addCustomTime method.
+   * 
+   */
+  const removeCustomTime = (id: string) => {
+    if (!timeline.value) return
+    timeline.value.removeCustomTime(id);
+  }
+
+  /**
+   * 
+   * Set a current time. This can be used for example to ensure that a client's 
+   * time is synchronized with a shared server time. Only applicable when 
+   * option showCurrentTime is true.
+   * 
+   */
+  const setCurrentTime = (time: Date | string) => {
+    if (!timeline.value) return
+    timeline.value.setCurrentTime(time);
+  }
+
+  /**
+   * 
+   * Adjust the time of a custom time bar.
+   * 
+   * @param time — The time the custom time bar should point to
+   * @param id — Id of the custom time bar, and is undefined by default.
+   * 
+   */
+  const setCustomTime = (time: Date | string, id: string) => {
+    if (!timeline.value) return
+    timeline.value.setCustomTime(time, id);
+  }
+
+  /**
+   * 
+   * Adjust the title attribute of a custom time bar.
+   * 
+   * @param title — The title shown when hover over time bar
+   * @param id — Id of the custom time bar, and is undefined by default.
+   * 
+   */
+  const setCustomTimeTitle = (title: string, id: string) => {
+    if (!timeline.value) return
+    timeline.value.setCustomTimeTitle(title, id);
+  }
+
+  onMounted(() => {
+    const element = resolveUnref(el)
+
+    if (!element) return
+
+    timeline.value = new Timeline(element, items.value)
+
+    if (groups.value && groups.value.length > 0) {
+      setGroups(groups.value)
+    }
```

Comment:
If `groups` exist, why not passing them in the Timeline constructor directly? Same for options?

I'm not sure what's the best in terms of performances and overall object handling: create once and for all a new Timeline, or adding properties after creation

Reply:
Yeh, I'll do some performance diffs and see what is more performant. It might not read as well though. Catch 22.

Reply:
It is not blocking anyway are both ways are correct.